tests/conftest.py

```diff
@@ -44,6 +46,523 @@ def pytest_logger_config(logger_config):
+                )
+
+    request.addfinalizer(finalizer)
+    return factory
+
+
+@pytest.fixture()
+def rbd_pvc_factory(rbd_storageclass_factory, pvc_factory):
+    """
+    Create a persistent Volume Claim factory. Calling this fixture creates new
+    RBD based PVC. For custom PVC provide 'storageclass' parameter.
+    """
+
+    def factory(
+        project=None,
+        storageclass=None,
+        custom_data=None,
+        status=constants.STATUS_BOUND
+    ):
+        """
+        Args:
+            project (object): ocs_ci.ocs.resources.ocs.OCS instance
+                of 'Project' kind.
+            storageclass (object): ocs_ci.ocs.resources.ocs.OCS instance
+                of 'StorageClass' kind.
+            custom_data (dict): If provided then PVC object is created
+                by using these data. Parameters `project` and `storageclass`
+                are not used but reference is set if provided.
+            status (str): If provided then factory waits for object to reach
+                desired state.
+
+        Returns:
+            object: helpers.create_pvc instance.
+        """
+        storageclass = storageclass or rbd_storageclass_factory()
+        return pvc_factory(
+            storageclass=storageclass,
+            project=project,
+            custom_data=custom_data,
+            status=status
+        )
+    return factory
+
+
+@pytest.fixture()
+def cephfs_pvc_factory(cephfs_storageclass_factory, pvc_factory):
+    """
+    Create a persistent Volume Claim factory. Calling this fixture creates new
+    CephFS based PVC. For custom PVC provide 'storageclass' parameter.
+    """
+
+    def factory(
+        project=None,
+        storageclass=None,
+        custom_data=None,
+        status=constants.STATUS_BOUND
+    ):
+        """
+        Args:
+            project (object): ocs_ci.ocs.resources.ocs.OCS instance
+                of 'Project' kind.
+            storageclass (object): ocs_ci.ocs.resources.ocs.OCS instance
+                of 'StorageClass' kind.
+            custom_data (dict): If provided then PVC object is created
+                by using these data. Parameters `project` and `storageclass`
+                are not used but reference is set if provided.
+            status (str): If provided then factory waits for object to reach
+                desired state.
+
+        Returns:
+            object: helpers.create_pvc instance.
+        """
+        storageclass = storageclass or cephfs_storageclass_factory()
+        return pvc_factory(
+            storageclass=storageclass,
+            project=project,
+            custom_data=custom_data,
+            status=status
+        )
+    return factory
+
+
+@pytest.fixture()
+def pod_factory(request):
+    """
+    Create a Pod factory. Calling this fixture creates new Pod.
+    For custom Pods provide 'pvc' parameter.
+    """
+    instances = []
+
+    def factory(pvc=None, custom_data=None, status=constants.STATUS_RUNNING):
```

Comment:
Are we sure we want to to default `pvc` to `None`? We aren't creating one if it isn't provided by the caller.

Reply:
I understand this as this is a general one and you have to pass pvc here if you don't provide custom_data.  So maybe worth to check on the line 470 that pvc objec is set and log the error. Anyway it will raise an exception AttributeError: 'NoneType' object has no attribute 'storageclass', right?

Reply:
I think if we are defaulting to `None` because we don't need it if we pass `custom_data` we should check for the existence of pvc if we don't provide `custom_data` and raise a proper exception if neither exist.